### PR TITLE
Fixes #398:-  Prototype for Media wall

### LINKED ISSUE #398: Adding module for social media walls
Related to #387 

Idea to add social media module which will have following components -

- [ ] Adding social media walls query component 
   
screen shot 2017-06-11 at 1 27 52 am

The query entered by user will make a call to search.json API so that interface is loaded lazily and all the statuses and media elements are loaded and rendered.

- [ ] Adding social media walls customization component

screen shot 2017-06-11 at 12 22 40 am

The colors can be customized by the user by choosing from the swatch. These colors will be the colors of the respective elements in the media wall Interface.

- [ ] Adding media wall component 

screen shot 2017-06-11 at 1 13 18 am

This is a Mock up for Media wall component. Since it is a media wall, more preference will be given to media elements. Setting option will allow user to customize colors and go in full screen so that media wall can be displayed on  different screens.

### PR DESCRIPTION
**Pull request by Achint08**
**Changes proposed in this pull request**
- Added necessary reducers and actions for loklak media walls.
  1.  To change Accent Color for media walls
  2.  To change Background Color for media walls
  3. To change Text Color for media walls
  4. To collect all media elements from Api Response to display on grid.
- Added Color customization 
- Added responsive images to loklak media wall
To avoid big PR, adding components in chunks.

Please review @hemantjadon .
**How to test**
Test link :- https://achint08.github.io/loklak_search/
You might find an option to disable "Unsafe scripts" by clicking on the shield/info shown on search bar. Disable it as api is on http. Search for something. You will find an option for media walls  
Click on "Display media wall" to show media wall This is currently a basic prototype for loklak media wall.

**Screenshot**

screen shot 2017-06-21 at 2 14 07 am

Fixes #398

### CODE REVIEW

**Comment by Achint08** [comment]
@hemantjadon 
 A) Yes. Look at this https://github.com/fossasia/labs.fossasia.org/issues/196 . Customization just for colors. That's it.
 B) For a single on going session, we need a state so that information can be kept.
 C) Yes. It will  be just an additional view.

**Comment by Achint08** [comment]
@hemantjadon Checkout now. A basic prototype of media walls. Any suggestion?

File: src/app/feed/feed-header/feed-header.component.html [hunk 1]
```diff
@@ -35,4 +35,7 @@
 				</ul>
 			<hr [style.left]="getCoordinates()">
 		</div>
+		<div class="mediaDisplay" *ngIf="(areResultsAvailable.length > 0) && (!resultsLoading)">
```
Comment (hemantjadon) [inline src/app/feed/feed-header/feed-header.component.html]: Make the `areResultsAvailable` as a boolean itself, why its length is checked?

File: src/app/feed/feed-header/feed-header.component.scss [hunk 2]
```diff
@@ -137,6 +137,12 @@
 				width:60px;
 			}
 	}
+
+	div.mediaDisplay {
```
Comment (hemantjadon) [inline src/app/feed/feed-header/feed-header.component.scss]: Use `kebab-case` for css classes and properties

File: src/app/reducers/media-wall.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
+import { createSelector } from 'reselect';
+import { Query, ReloactionAfterQuery } from '../models/query';
+
+import * as apiAction from '../actions/api';
+import * as mediaWallAction from '../actions/media-wall';
+import { ApiResponse } from '../models/api-response';
+
+
+export interface State {
+	MediaWallsColor;
```
Comment (hemantjadon) [inline src/app/reducers/media-wall.ts]: Use `MediaWallColor`, not plural

File: src/app/media-wall/media-wall.component.scss [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+	.title {
+		position:absolute;
+		top:1.5vh;
+		left:5%;
+		width:90%;
+		height:4vh;
+		text-align:center;
+		display:block;
+  	font-family: 'Montserrat', sans-serif !important;
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall.component.scss]: Indentation error I guess

File: src/app/media-wall/media-wall.component.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import { Component,
+					OnInit, AfterViewInit, OnDestroy,
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall.component.ts]: Indentation Error Maybe

**Comment by Achint08** [comment]
@djmgit @singhpratyush @kavithaenair @vibhcool Please review :)

**Comment by mariobehling** [comment]
Which issue is this related to?

**Comment by Achint08** [comment]
@mariobehling This is a basic prototype of loklak media wall. This is a base for loklak media wall.
All the actions, services are added for loklak media wall to run. It is related to #398 .

**Comment by hemantjadon** [comment]
@Achint08 Please fix the merge conflicts then this PR is good to go 👍

**Comment by mariobehling** [comment]
@Achint08 Screenshots, Link to issue and link to a test site are not provided. Please follow best practices!

**Comment by mariobehling** [comment]
Travis is failing in this PR. Please fix it.

**Comment by Achint08** [comment]
@mariobehling I deleted the test link by mistake. Updated it. Do have a look once.
Check on bigger screens/smaller screens if image adjusts itself to every type of screen.
The travis is failing due to previous PR merged. Looking into that.

**Comment by mariobehling** [comment]
We had discussed this feature a number of times and it was mentioned that it should be integrated into the existing application -> as a button "display this search result as a media wall". This is a very simple requirement and we don't need a black page. A simple example is already on the apps page: http://apps.loklak.org/details.html?q=social-wall 

I am unclear why you are not following this.

First step: After the user makes a search, on the search result pags add a text button "Views". When the user clicks on it an option appears "Media Wall". When the user clicks this button another tab opens showing the search results as a media wall in a simple style as http://apps.loklak.org/details.html?q=social-wall
screenshot from 2017-06-20 23-10-49

**Comment by mariobehling** [comment]
Follow best practices!

> All the actions, services are added for loklak media wall to run. It is related to #398 .

Add this issue to the description!

**Comment by Achint08** [comment]
@mariobehling I have done like that only. Checkout here.
https://achint08.github.io/loklak_search/ .
 

screen shot 2017-06-21 at 1 27 12 pm

However, my idea about media wall is to give more emphasis on media so it will have a UI like this.

screen shot 2017-06-21 at 1 30 07 pm

File: src/app/feed/feed-header/feed-header.component.scss [hunk 6]
```diff
@@ -174,4 +180,8 @@
 		left: 20px !important;
```
Comment (vibhcool) [inline src/app/feed/feed-header/feed-header.component.scss]: Could you remove `!important;`, if it is not needed. It seems to be bad practice. And the first answer in this link is too comical 😂  
https://stackoverflow.com/questions/9245353/what-does-important-in-css-mean

**Comment by mariobehling** [comment]
@Achint08 It does not make sense if each single feature of the app has a different design. Please follow the design of the app. Implement a simple first version based on the existing work in the FOSSASIA community.

**Pull request closed without merging**